kratos/tests/global_pointers/test_global_pointer.cpp

```diff
@@ -0,0 +1,174 @@
+  private:
+    TestClass() {}
+
+  public:
+    TestClass(int magicNumber) { this->mMagicNumber = magicNumber; }
+    TestClass(const TestClass & rOther) { this->mMagicNumber = rOther.mMagicNumber; }
+
+    ~TestClass() {}
+
+    int getVar() const { return this->mMagicNumber; }
+
+    void setVar(int magicNumber) { this->mMagicNumber = magicNumber; }
+
+    int mMagicNumber;
+};
+
+KRATOS_TEST_CASE_IN_SUITE(GlobalPointerCreateRaw, KratosCoreFastSuit)
+{
+  int sampleVar = 1337;
+
+	auto fromRaw = GlobalPointer<int>(&sampleVar);
+
+  KRATOS_CHECK_EQUAL(*fromRaw, sampleVar);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GlobalPointerCreateConstRaw, KratosCoreFastSuit)
+{
+  const int sampleVar = 1337;
+
+	auto fromRaw = GlobalPointer<const int>(&sampleVar);
+
+  KRATOS_CHECK_EQUAL(*fromRaw, sampleVar);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GlobalPointerModifyRaw, KratosCoreFastSuit)
+{
+  int sampleVar = 1337;
+  int newVal = 42;
+
+	auto fromRaw = GlobalPointer<int>(&sampleVar);
+  *fromRaw = newVal;
+
+  KRATOS_CHECK_EQUAL(*fromRaw, newVal);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GlobalPointerCreateClass, KratosCoreFastSuit)
+{
+  TestClass sampleVar(1337);
+
+	auto fromRaw = GlobalPointer<TestClass>(&sampleVar);
+
+  KRATOS_CHECK_EQUAL(fromRaw->getVar(), sampleVar.getVar());
+  KRATOS_CHECK_EQUAL((*fromRaw).getVar(), sampleVar.getVar());
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GlobalPointerCreateConstClass, KratosCoreFastSuit)
+{
+  const TestClass sampleVar(1337);
+
+	auto fromRaw = GlobalPointer<const TestClass>(&sampleVar);
+
+  KRATOS_CHECK_EQUAL(fromRaw->getVar(), sampleVar.getVar());
+  KRATOS_CHECK_EQUAL((*fromRaw).getVar(), sampleVar.getVar());
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GlobalPointerModifyClass, KratosCoreFastSuit)
+{
+  TestClass sampleVar(1337);
+
+	auto fromRaw = GlobalPointer<TestClass>(&sampleVar);
+
+  fromRaw->setVar(42);
+  sampleVar.setVar(42);
+
+  KRATOS_CHECK_EQUAL(fromRaw->getVar(), sampleVar.getVar());
+  KRATOS_CHECK_EQUAL((*fromRaw).getVar(), sampleVar.getVar());
+}
+
+// Test global_ptr<shared_ptr>
+KRATOS_TEST_CASE_IN_SUITE(GlobalPointerCreateBoostSharedPtr, KratosCoreFastSuit)
+{
+  typedef boost::shared_ptr<TestClass> BoostPtrType;
+
+  auto sampleVar = BoostPtrType(new TestClass(1337));
+	auto fromBoost = GlobalPointer<TestClass>(sampleVar);
+
+  KRATOS_CHECK_EQUAL(fromBoost->getVar(), sampleVar->getVar());
+  KRATOS_CHECK_EQUAL((*fromBoost).getVar(), sampleVar->getVar());
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GlobalPointerModifyBoostSharedPtr, KratosCoreFastSuit)
+{
+  typedef boost::shared_ptr<TestClass> BoostPtrType;
+
+  auto sampleVar = BoostPtrType(new TestClass(1337));
+	auto fromBoost = GlobalPointer<TestClass>(sampleVar);
+
+  fromBoost->setVar(42);
+  sampleVar->setVar(42);
+
+  KRATOS_CHECK_EQUAL(fromBoost->getVar(), sampleVar->getVar());
+  KRATOS_CHECK_EQUAL((*fromBoost).getVar(), sampleVar->getVar());
+}
+
+/// Parallel tests
+
+#ifdef KRATOS_USING_MPI
+KRATOS_TEST_CASE_IN_SUITE(GlobalPointerGatherRaw, KratosCoreFastSuit)
```

Comment:
I think we should add a macro for MPI test cases which deos the enable disable. But it would be subject of another branch. It is ok as you did now.